src/Sylius/Component/Core/Updater/UnpaidOrdersStateUpdater.php

```diff
@@ -29,24 +30,43 @@ final class UnpaidOrdersStateUpdater implements UnpaidOrdersStateUpdaterInterfac
     /** @var string */
     private $expirationPeriod;
 
+    /** @var LoggerInterface|null */
+    private $logger;
+
     /**
      * @param string $expirationPeriod
      */
     public function __construct(
         OrderRepositoryInterface $orderRepository,
         Factory $stateMachineFactory,
-        $expirationPeriod
+        $expirationPeriod,
+        LoggerInterface $logger = null
     ) {
         $this->orderRepository = $orderRepository;
         $this->stateMachineFactory = $stateMachineFactory;
         $this->expirationPeriod = $expirationPeriod;
+        if (null === $logger) {
+            @trigger_error(
+                'Not passing a logger is deprecated since 1.7',
+                \E_USER_DEPRECATED
+            );
+        }
+
+        $this->logger = $logger;
     }
 
     public function cancel(): void
     {
         $expiredUnpaidOrders = $this->orderRepository->findOrdersUnpaidSince(new \DateTime('-' . $this->expirationPeriod));
         foreach ($expiredUnpaidOrders as $expiredUnpaidOrder) {
-            $this->cancelOrder($expiredUnpaidOrder);
+            try {
+                $this->cancelOrder($expiredUnpaidOrder);
+            } catch (\Exception $e) {
+                $this->logger && $this->logger->error(
```

Comment:
Can we make explicit if statement to improve readability?

Reply:
Hey @lchrusciel,

I just added an explicit `instance of` check.